src/_pytest/pytester.py

```diff
@@ -1213,7 +1220,7 @@ def runpytest_subprocess(self, *args, **kwargs):
         if plugins:
             args = ("-p", plugins[0]) + args
         args = self._getpytestargs() + args
-        return self.run(*args, timeout=kwargs.get("timeout"))
+        return self.run(*args, timeout=timeout)
```

Comment:
Note that this test only works accidentally, and is not testing the original issue anymore (see https://github.com/pytest-dev/pytest/issues/5084), but that should not hold this PR back.